include/message_filters/cache.hpp

```diff
@@ -66,20 +66,37 @@ class Cache : public SimpleFilter<M>
   typedef MessageEvent<M const> EventType;
 
   template<class F>
-  explicit Cache(F & f, unsigned int cache_size = 1)
+  explicit Cache(F & f, unsigned int cache_size = 1, bool allow_headerless = false)
```

Comment:
I think this is fine from an API point of view, but I think there may be some ABI implications around adding a new parameter with a default value.

Any downstream consumers of message filters would need to recompile, which is generally not a problem, but means that this could not be backported.

Reply:
Maybe it is better for backward compatibility to add two overloaded constructors with this signature, and new functionality, and keep the old interface intact?

Reply:
As @mjcarroll mentioned, If we want to backport this feature we need to create a new constructor. Otherwise we will break ABI compatibility.

You can find here a list of ABI breaking changes https://acodersjourney.com/20-abi-breaking-changes/

Reply:
you can try with this new signature

```cpp
explicit Cache(F & f, unsigned int cache_size, bool allow_headerless)
```

Reply:
Added new constructors. Seems like it is working now.